tests/lib/rules/require-typed-ref.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * @author Ivan Demchuk <https://github.com/Demivan>
+ */
+'use strict'
+
+const RuleTester = require('eslint').RuleTester
+const rule = require('../../../lib/rules/require-typed-ref')
+
+const tester = new RuleTester({
+  parser: require.resolve('@typescript-eslint/parser'),
+  parserOptions: { ecmaVersion: 2020, sourceType: 'module' }
+})
+
+// Note: Need to specify filename for each test,
+// as only TypeScript files are being checked
+tester.run('require-typed-ref', rule, {
+  valid: [
+    {
+      filename: 'test.ts',
+      code: `
+        import { shallowRef } from 'vue'
+        const count = shallowRef(0)
+      `
+    },
+    {
+      filename: 'test.ts',
+      code: `
+        import { ref } from 'vue'
+        const count = ref<number>()
+      `
+    },
+    {
+      filename: 'test.ts',
+      code: `
+        import { ref } from 'vue'
+        const count = ref<number>(0)
+      `
+    },
+    {
+      filename: 'test.ts',
+      code: `
+        import { ref } from 'vue'
+        const counter: Ref<number | undefined> = ref()
+      `
+    },
+    {
+      filename: 'test.ts',
+      code: `
+        import { ref } from 'vue'
+        const count = ref(0)
+      `
+    },
+    {
+      filename: 'test.ts',
+      code: `
+        import { ref } from 'vue'
+        function useCount() {
+          return {
+            count: ref<number>()
+          }
+        }
+      `
+    },
+    {
+      filename: 'test.vue',
+      parser: require.resolve('vue-eslint-parser'),
+      code: `
+        <script setup>
+          import { ref } from 'vue'
+          const count = ref()
+        </script>
+      `
+    },
+    {
+      filename: 'test.js',
+      code: `
+        import { ref } from 'vue'
+        const count = ref()
+      `
+    }
```

Comment:
Please also add tests (and fix the rule if necessary) for the Options API usage (`defineComponent` with a `setup` function).

Reply:
Done and fixed other comments